### ISSUE TITLE
How to implement role-based security in a WPF application

### ISSUE THREAD

**Issue by MornyDK** [dotnet/docs#20493]
The start of [this](https://docs.microsoft.com/en-us/dotnet/standard/security/role-based-security) article states: "Roles are often used in financial or business applications to enforce policy".
I'm developing exactly this type of application, but are unable to find samlps that shows how to implement role or policy based security in a WPF application on .Net Core.
On .Net Framework CAS (Code Access Security) can be used to trigger the permission check by decorating methods with attributes, as many samples on the internet shows, e.g. [this one](https://blog.magnusmontin.net/2013/03/24/custom-authorization-in-wpf/)

How can role-based security be implemented in a WPF application so that checks are triggered, preferably by attribute decorations (link to samlles will be much appreciated)?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6a5f30cb-fa02-1f7a-88f6-a387b77324da
* Version Independent ID: 1c902e29-43be-685d-81ff-d37048db5a8e
* Content: [Role-Based Security](https://docs.microsoft.com/en-us/dotnet/standard/security/role-based-security)
* Content Source: [docs/standard/security/role-based-security.md](https://github.com/dotnet/docs/blob/master/docs/standard/security/role-based-security.md)
* Product: **dotnet**
* Technology: **dotnet-standard**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

**Comment by IEvangelist** [comment]
Hey @OliaG - do you know of any example WPF apps that we could incorporate into docs that include role-based security?

**Comment by adegeo** [comment]
I don't know of any. There are some solutions suggested on stack overflow: https://stackoverflow.com/search?q=role-based+security+WPF

You can also browse the WPF samples repository: https://github.com/Microsoft/WPF-Samples but I checked here and didn't see anything.

I think you'll have to either come up with your own solution or find a tutorial such as https://social.technet.microsoft.com/wiki/contents/articles/25726.wpf-implementing-custom-authentication-and-authorization.aspx

@IEvangelist You need to ping Olia in email/teams. She's doesn't really use github notifications (I don't think) 😄

**Comment by IEvangelist** [comment]
Hi @adegeo - I messaged her on Teams already. She too didn't know of any. I did find this https://docs.microsoft.com/en-us/dotnet/desktop/wpf/security-wpf?view=netframeworkdesktop-4.8. I also found a few things of SO too, might borrow some of the concepts there but it seems it is as straight forward as walking up to the `Thread.CurrentPrincipal` and asking it if the user is in a role (based on [this](https://stackoverflow.com/a/1020303/2410379)).

```csharp
Thread.CurrentPrincipal.IsInRole("Administrator")
```
